test_lib/cql_types.py

```diff
@@ -0,0 +1,200 @@
+import random
+
+
+__ALL__ = ['CSQLColumnType', 'ALL_COLUMN_TYPES', 'NOT_EMBEDDABLE_COLUMN_TYPES', 'COLLECTION_COLUMN_TYPES']
+
+
+ALL_COLUMN_TYPES = ('ascii', 'blob', 'counter', 'decimal', 'duration', 'int', 'map', 'smallint', 'time',
+                    'timeuuid', 'uuid', 'varint', 'bigint', 'boolean', 'date', 'double', 'float', 'inet', 'list', 'set',
+                    'text', 'timestamp', 'tinyint', 'varchar')
+NOT_EMBEDDABLE_COLUMN_TYPES = ['duration', 'counter']
+COLLECTION_COLUMN_TYPES = ['set', 'list', 'map']
+
+
+class CSQLColumnType:
+    maintype = None
+    subtype1 = None
+    subtype2 = None
+    parent = None
+    _registered_classes = {}
+
+    def __init_subclass__(cls, **kwargs):
+        if cls.maintype is None:
+            raise ValueError("<%s> maintype should be defined" % (str(cls),))
+        if cls._registered_classes.get(cls.maintype, None) is not None:
+            raise RuntimeError("<%s> maintype %s is already registered" % (str(cls), cls.maintype))
+        cls._registered_classes[cls.maintype] = cls
+
+    def __new__(cls, **kwargs):
```

Comment:
this is bad coding practice: Read more here: https://stackoverflow.com/questions/6465686/how-to-implement-a-required-property-in-python